pmlc/dialect/tile/ir/ops.cc

```diff
@@ -387,9 +387,27 @@ struct SymbolicContractionCanonicalizer
     auto sourceType = op.result().getType().cast<RankedTensorType>();
     auto resultType = RankedTensorType::get(shape, sourceType.getElementType());
     if (!resultType.hasStaticShape()) {
-      return failure();
+      if (resultType == sourceType) {
+        return failure();
+      }
+      // Can't rewrite to a non-symbolic contraction, but we can at least update
+      // the result type
+      auto newOp = rewriter.create<SymbolicContractionOp>(
+          op.getLoc(), resultType, op.init(), op.cons(), op.size(), op.sink(),
+          op.srcs(), op.agg(), op.combo(), rewriter.getUnitAttr(),
+          rewriter.getStringAttr(op.name().getValueOr("")));
+      // TODO: This seems like not the right way to handle no_reduce
+      if (op.no_reduce().hasValue()) {
+        newOp.setAttr("no_reduce", rewriter.getUnitAttr());
+      } else {
+        newOp.removeAttr("no_reduce");
+      }
+      rewriter.replaceOp(op, newOp.result());
+      util::UpdateFuncOpType(newOp.getOperation());
+      return success();
     }
 
+    // TODO: Do I need to verify foldability first?
```

Comment:
Yes, we can't fold min/max.

Reply:
That seems to me like it means I _don't_ need to move the foldability check above the partially static shape code. The partially static shape code is just rewriting the result type, not changing from a symbolic contraction to a non-symbolic contraction. So it seems like inability to fold min/max shouldn't matter. Or am I missing something?

Reply:
OK, let's remove the comment then.